### PR TITLE
Bug 1286858, Documented the infra region

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1286858

@adellape PTAL. Thanks!

### CODE REVIEW

File: install_config/install/advanced_install.adoc [hunk 1]
```diff
@@ -315,6 +315,23 @@ node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
...
+====
+----
+# default selectors for router and registry services
+# openshift_router_selector='region=infra'
+# openshift_registry_selector='region=infra'
+----
+====
+
+The default router and registry are automatically deployed. The default label is
+`*region=infra*`. In all other aspects, this is a normal node. Not all nodes are
```
Comment (brenton) [inline install_config/install/advanced_install.adoc]: The router / registry will only be automatically deploy if Nodes exist with labels that match the selector settings above.  Here's an example of how ansible labels can apply labels to Nodes: https://github.com/openshift/openshift-ansible/blob/master/inventory/byo/hosts.ose.example#L289

Comment (ahardin-rh) [inline install_config/install/advanced_install.adoc]: @brenton Thanks! Made some edits. PTAL and let me know if it's clearer.

File: install_config/install/advanced_install.adoc [hunk 2]
```diff
@@ -315,6 +315,34 @@ node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
 ----
 ====
 
+Nodes should have a label signifying the router. The registry should be
+installed on them. For example:
```
Comment (brenton) [inline install_config/install/advanced_install.adoc]: This part sounds confusing to me.  Nodes have labels that often indicate their region.  A convention is to set the 'region=infra' label for the router and registry selector.

**Comment by ahardin-rh** [comment]
@bfallonf @tpoitras @tnguyen-rh PTAL as well. Thanks!

File: install_config/install/advanced_install.adoc [hunk 3]
```diff
@@ -315,6 +315,34 @@ node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
...
+----
+# default selectors for router and registry services
+# openshift_router_selector='region=infra'
+# openshift_registry_selector='region=infra'
+----
+====
+
+The default label is `*region=infra*`. In all other aspects, this is a normal
+node. Not all nodes are running mission-critical containers. While a node
+labeled `*dev*` might be rebooted or deleted at any time, the `*infra*` nodes
```
Comment (tnguyen-rh) [inline install_config/install/advanced_install.adoc]: A label comprises key and value, so "labelled *dev*" is incomplete (likewise, naked *infra*).  For the latter, i suppose the key would be *region* (thus, "the `*region=infra*` nodes").  Maybe the same key is used for *dev*, but i'm guessing not.

**Comment by tnguyen-rh** [comment]
@ahardin-rh One small clarification required, otherwise LGTM.

**Comment by ahardin-rh** [comment]
@tnguyen-rh Fixed. Thanks!

File: install_config/install/advanced_install.adoc [hunk 4]
```diff
@@ -315,6 +315,34 @@ node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
 ----
 ====
 
+Nodes have labels that often indicate their region. A convention is to set the
+`*region=infra*` label for the router and registry selector. For example:
```
Comment (bfallonf) [inline install_config/install/advanced_install.adoc]: @ahardin-rh I think the default setting is a little confused here. Above, you're telling the reader to set the region=infra label for the router and registry, but below, you're telling them that nodes have the label by default. Should the below say "All new nodes will have the *region=infra* label by default"?

**Comment by bfallonf** [comment]
@ahardin-rh One question, but otherwise looks good!

File: install_config/install/advanced_install.adoc (hunk 4 above)
Comment (brenton) [inline install_config/install/advanced_install.adoc]: Nodes only receive labels today if you explicitly add them in the inventory using `openshift_node_labels` as described below.  `region=infra` is the default selector and if you create Nodes with that selector then the router and registry will be installed.

File: install_config/install/advanced_install.adoc [hunk 5]
```diff
@@ -315,6 +315,34 @@ node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
...
+====
+----
+# default selectors for router and registry services
+# openshift_router_selector='region=infra'
+# openshift_registry_selector='region=infra'
+----
+====
+
+The default label is `*region=infra*`. In all other aspects, this is a normal
+node. Not all nodes are running mission-critical containers. While a node
```
Comment (brenton) [inline install_config/install/advanced_install.adoc]: This is probably the section that @bfallonf is getting confused by.  We're talking about label selectors and then the phrase 'this is a normal node' is used.  That seems out of place.  I would simply say something like, "If `openshift_node_labels` on a Node includes a label that matches the router or registry selector those services will be deployed by ansible."

Comment (brenton) [inline install_config/install/advanced_install.adoc]: Looking again I would say this paragraph could simply be removed.  The following one covers everything I just mentioned.

**Comment by ahardin-rh** [comment]
@bfallonf @brenton Thanks! Made some updates per your comments.

File: install_config/install/advanced_install.adoc [hunk 6]
```diff
@@ -315,6 +315,30 @@ node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
...
+selector. If you create nodes with that selector, then the router and registry
+will be installed.
+
+====
+----
+# default selectors for router and registry services
+# openshift_router_selector='region=infra'
+# openshift_registry_selector='region=infra'
+----
+====
```
Comment (bfallonf) [inline install_config/install/advanced_install.adoc]: @ahardin-rh Sorry, i should have picked up on this yesterday, but is the "default selectors for router and registry services" statement supposed to be a uncommented meant for context? The other two lines are commands for the reader to run, right? If this is all true, the I think that line can probably be removed. The para above is stating that anyway.

**Comment by bfallonf** [comment]
@ahardin-rh Sorry, one more comment :pineapple:

File: install_config/install/advanced_install.adoc (hunk 6 above)
Comment (brenton) [inline install_config/install/advanced_install.adoc]: Those lines are likely taken directly from the configuration file.  It's common for configuration files to show the defaults by comments.  That way should an admin wish to change them they simply uncomment the line and then change the value.  If they don't change the value then there really isn't any point to uncomment them.  Personally, I would leave these lines as is.

Comment (ahardin-rh) [inline install_config/install/advanced_install.adoc]: Thanks, @bfallonf and @brenton  I will leave the lines as is per Brenton's comment.

**Comment by ahardin-rh** [comment]
@brenton @bfallonf made a few edits per conversation with @adellape. This should be clearer/less redundant now. PTAL. Thanks!

**Comment by ahardin-rh** [comment]
@brenton Did openshift_router_selector and openshift_registry_selector start working with 3.1? Thanks!

**Comment by brenton** [comment]
@ahardin-rh, I know we didn't have them in 3.0.  How exact do you need the answer to be?  I'm not sure if it was 3.1 or 3.1.1.

**Comment by ahardin-rh** [comment]
@brenton perfect. Thanks for clarifying!

**Comment by ahardin-rh** [comment]
[rev_history]
|link:../install_config/install/advanced_install.html[Advanced Installation]
|Added information about *region=infra* to the link:../install_config/install/advanced_install.html#configuring-node-host-labels[Configuring Node Host Labels] section and added `*openshift_router_selector*` and `*openshift_registry_selector*` to the link:../install_config/install/advanced_install.html#configuring-host-variables[Host Variables] table.
%

**Comment by anpingli** [comment]
The installing Ansible version  should be 1.8.4 to 1.9.x.  I don't think we support Ansible 2.0 and later now.

**Pull request merged**